src/routes/subscriptions/subscription.controller.spec.ts

```diff
@@ -0,0 +1,165 @@
+      .useModule(TestCacheModule)
+      .overrideModule(RequestScopedLoggingModule)
+      .useModule(TestLoggingModule)
+      .overrideModule(NetworkModule)
+      .useModule(TestNetworkModule)
+      .compile();
+
+    emailDatasource = moduleFixture.get(IEmailDataSource);
+
+    app = await new TestAppProvider().provide(moduleFixture);
+    await app.init();
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  it('deletes category successfully', async () => {
+    const subscriptionKey = faker.word.sample();
+    const subscriptionName = faker.word.sample(2);
+    const token = faker.string.uuid();
+    const subscriptions = [
+      {
+        key: subscriptionKey,
+        name: subscriptionName,
+      },
+    ] as Subscription[];
+    emailDatasource.unsubscribe.mockResolvedValueOnce(subscriptions);
+
+    await request(app.getHttpServer())
+      .delete(`/v1/subscriptions/?category=${subscriptionKey}&token=${token}`)
+      .expect(200)
+      .expect({});
+
+    expect(emailDatasource.unsubscribe).toHaveBeenCalledWith({
+      categoryKey: subscriptionKey,
+      token: token,
+    });
+    expect(emailDatasource.unsubscribeAll).toHaveBeenCalledTimes(0);
+  });
+
+  it('validates uuid format when deleting category', async () => {
+    const subscriptionKey = faker.word.sample();
+    const token = faker.string.hexadecimal();
+
+    await request(app.getHttpServer())
+      .delete(`/v1/subscriptions/?category=${subscriptionKey}&token=${token}`)
+      .expect(400)
+      .expect({
+        message: 'Validation failed (uuid is expected)',
+        error: 'Bad Request',
+        statusCode: 400,
+      });
+
+    expect(emailDatasource.unsubscribe).toHaveBeenCalledTimes(0);
+    expect(emailDatasource.unsubscribeAll).toHaveBeenCalledTimes(0);
+  });
+
+  it('deleting category is not successful', async () => {
+    const subscriptionKey = faker.word.sample();
+    const token = faker.string.uuid();
+    emailDatasource.unsubscribe.mockRejectedValueOnce(new Error('some error'));
+
+    await request(app.getHttpServer())
+      .delete(`/v1/subscriptions/?category=${subscriptionKey}&token=${token}`)
+      .expect(500)
+      .expect({ code: 500, message: 'Internal server error' });
+  });
+
+  it('deletes all categories successfully', async () => {
+    const subscriptionKey = faker.word.sample();
+    const subscriptionName = faker.word.sample(2);
+    const token = faker.string.uuid();
+    const subscriptions = [
+      {
+        key: subscriptionKey,
+        name: subscriptionName,
+      },
+    ] as Subscription[];
+    emailDatasource.unsubscribeAll.mockResolvedValueOnce(subscriptions);
+
+    await request(app.getHttpServer())
+      .delete(`/v1/subscriptions/all?token=${token}`)
+      .expect(200)
+      .expect({});
+
+    expect(emailDatasource.unsubscribeAll).toHaveBeenCalledWith({
+      token: token,
+    });
+    expect(emailDatasource.unsubscribe).toHaveBeenCalledTimes(0);
```

Comment:
Nit: can we also check here that the database updated successfully?

Reply:
See https://github.com/safe-global/safe-client-gateway/pull/1037#discussion_r1459039745